plan/point_select_plan.go

```diff
@@ -0,0 +1,435 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/parser/opcode"
+	"github.com/pingcap/tidb/privilege"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tipb/go-tipb"
+)
+
+// PointSelectPlan is a fast plan for point select.
+type PointSelectPlan struct {
+	basePlan
+	schema      *expression.Schema
+	TblInfo     *model.TableInfo
+	IndexInfo   *model.IndexInfo
+	Handle      int64
+	IndexValues []types.Datum
+	ctx         sessionctx.Context
+	stats       *statsInfo
+}
+
+type nameValuePair struct {
+	colName string
+	value   types.Datum
+}
+
+// Schema implements the Plan interface.
+func (fp *PointSelectPlan) Schema() *expression.Schema {
+	return fp.schema
+}
+
+// attach2Task makes the current physical plan as the father of task's physicalPlan and updates the cost of
+// current task. If the child's task is cop task, some operator may close this task and return a new rootTask.
+func (fp *PointSelectPlan) attach2Task(...task) task {
+	return nil
+}
+
+// ToPB converts physical plan to tipb executor.
+func (fp *PointSelectPlan) ToPB(ctx sessionctx.Context) (*tipb.Executor, error) {
+	return nil, nil
+}
+
+// ExplainInfo returns operator information to be explained.
+func (fp *PointSelectPlan) ExplainInfo() string {
+	return "Point_Select"
+}
+
+// getChildReqProps gets the required property by child index.
+func (fp *PointSelectPlan) getChildReqProps(idx int) *requiredProp {
+	return nil
+}
+
+// StatsInfo will return the statsInfo for this plan.
+func (fp *PointSelectPlan) statsInfo() *statsInfo {
+	return fp.stats
+}
+
+// StatsCount will return the the count of statsInfo for this plan.
+func (fp *PointSelectPlan) StatsCount() float64 {
+	return fp.stats.count
+}
+
+// Children gets all the children.
+func (fp *PointSelectPlan) Children() []PhysicalPlan {
+	return nil
+}
+
+// SetChildren sets the children for the plan.
+func (fp *PointSelectPlan) SetChildren(...PhysicalPlan) {}
+
+// ResolveIndices resolves the indices for columns. After doing this, the columns can evaluate the rows by their indices.
+func (fp *PointSelectPlan) ResolveIndices() {}
+
+func tryFastPlan(ctx sessionctx.Context, node ast.Node) Plan {
+	if PreparedPlanCacheEnabled {
+		// plan cache not supported yet.
```

Comment:
Is there any plan to support fast-path with plan cache?

Reply:
No

Reply:
I think the comment could be improved.